app/components/subscription/prices.tsx

```diff
@@ -47,56 +55,97 @@ export const Price = ({
   const features = price.product.metadata.features?.split(",") || [];
+
+  // icons to be mapped with different plans
+  const plansIconsMap: { [key: string]: JSX.Element } = {
+    free: <SingleLayerIcon />,
+    tier_1: <DoubleLayerIcon />,
+    tier_2: <MultiLayerIcon />,
+  };
+
+  function shortenIntervalString(str: string): string {
+    // Using a regular expression to replace 'year' with 'yr' and 'month' with 'mo'
+    return str.replace(/\b(year|month)\b/g, (match: string) => {
+      if (match === "year") return "yr";
+      else if (match === "month") return "mo";
+      else return str;
+    });
+  }
+
   return (
-    <div
-      key={price.id}
-      className={tw(
-        " w-full border-2 border-white bg-gray-100 p-4 hover:border-primary-200",
-        activePlan?.id === price.id &&
-          "border-2 border-primary-500 hover:border-primary-500",
-        !activeSubscription &&
-          price.id === "free" &&
-          "border-2 border-primary-500"
-      )}
-    >
-      <div>
-        <h3>{price.product.name}</h3>
-      </div>
-      {price.unit_amount != null ? (
-        <>
-          <div className="text-xl">
-            {price.unit_amount / 100} {price.currency}{" "}
-            {price.recurring ? (
-              <span className="text-[14px]">
-                per {price.recurring.interval}
-              </span>
+    <div className="subscription-plan mb-12 w-full">
+      <div
+        className={tw(
+          "mb-8 rounded-2xl border p-8",
+          activePlan?.id === price.id ||
+            (!activeSubscription && price.id === "free")
+            ? "border-primary-500 bg-primary-50"
+            : "bg-white"
+        )}
+        key={price.id}
+      >
+        <div className="text-center">
+          <div className="mb-5 inline-flex items-center justify-center rounded-full border-[5px] border-solid border-primary-50 bg-primary-100 p-1.5 text-primary">
+            <i className=" inline-flex min-h-[20px] min-w-[20px] items-center justify-center">
+              {price.product.metadata.shelf_tier
+                ? plansIconsMap[price.product.metadata.shelf_tier]
+                : plansIconsMap["free"]}
+            </i>
+          </div>
+          <div className="mb-3 flex items-center justify-center gap-2">
+            <h2 className=" text-xl font-semibold text-primary-700">
+              {price.product.name}
+            </h2>
+            {activePlan?.id === price.id ||
+            (!activeSubscription && price.id === "free") ? (
+              <div className="rounded-2xl bg-primary-50 px-2 py-0.5 text-[12px] font-medium text-primary-700 mix-blend-multiply">
+                Current
+              </div>
             ) : null}
           </div>
-        </>
-      ) : null}
-      <div>
-        <i>{price.product.metadata.slogan}</i>
+          {price.unit_amount != null ? (
+            <div className=" mb-3 text-4xl font-semibold text-gray-900">
+              {price.currency === "usd"
```

Comment:
I think there is a better way to do it. You should be able to use `toLocaleString` and pass a currency and this will format it accordingly by placing the correct char in front or behind.